pkg/repository/chartmuseum.go

```diff
@@ -180,6 +181,14 @@ func (c *chartmuseum) Poll() {
 		c.logger.Error(err, "try to update repository, but failed to get the latest version.", "readyCond", readyCond, "syncCond", syncCond)
 	} else {
 		iDeepCopy := i.DeepCopy()
+		if syncCond.LastSuccessfulTime.IsZero() {
```

Comment:
add a comment here to let people know what you are trying to do.

`set sync.Cond.LastSuccessfulTime` to  previous `LastSuccessfulTime` if it exists.

Reply:
No. The `SetConditions` function is heavily used, and I modified this function, which may affect other places where it is called.